### PR TITLE
Autogenerate events resource

### PR DESCRIPTION
**Pull request by agualis**
I just used the auto-generator to create this `events-resource.js` file that contains both `timeline events` (cc @rensftw ) and `ruleset events` (cc @mattsanf ).

I used the custom names that you can see although, in this concrete case, as they  new functions, I could also use more canonical names instead (from the `operationId`s in api-definitions) as I discussed with Matt in Slack today. 

@Irena I think that the code related with your RFC is equivalent to [this PR ](https://github.com/Rebilly/rebilly-js-sdk/pull/277)that you already created, so feel free to merge that one if you need it.

@mattsanf Also note that I changed `Ruleset` by `RuleSet` in the names. I can keep the original name or using the new one and fixing recomm usages when pointing to the new version. 

And one last detail: I see that the current manual code has  `createRules` and `updateRules` both using the same `put` request. That case was not contemplated in the code generator. Can we merge them into one unique method with a custom name or do we need to add that special case to the generator?

### CODE REVIEW

**Review by rensftw — APPROVED** [review]
LGTM

**Comment by rensftw** [comment]
@agualis I merged the [older PR](https://github.com/Rebilly/rebilly-js-sdk/pull/277) because I need the changes, but this PR looks good to me.

>I see that the current manual code has `createRules` and `updateRules` both using the same put request. That case was not contemplated in the code generator. Can we merge them into one unique method with a custom name or do we need to add that special case to the generator?

It looks like `createRules` is not used anywhere on Recomm, so I think we can remove it altogether and bump the major version.

**Comment by rensftw** [comment]
(Also, I resolved the conflicts myself - hope I didn't mess withh the autogenerated code)

**Review by rensftw — APPROVED** [review]

**Review by rensftw — APPROVED** [review]

**Review by rensftw — CHANGES_REQUESTED** [review]
There seems to be an issue with this endpoint:
`events@{eventType}@rules@versions@{version}.yaml`

**Comment by agualis** [comment]
> There seems to be an issue with this endpoint:
> `events@{eventType}@rules@versions@{version}.yaml`

That yaml is related with `getRulesVersionDetail` function. The only difference with the original function is that the auto-generated one can accept  optional parameters (`fields` and `expand`).

**Comment by rensftw** [comment]
@agualis It looks like I pasted the wrong thing yesterday :( 
This is the specific URL that is failing for me: `https://api-sandbox.rebilly.com/events/customer-updated/rules/drafts/20cb05da-0109-4958-b899-5739eb19f315`

So I guess it would be this schema: `openapi/paths/events@{eventType}@rules@drafts@{id}.yaml`

File: src/resources/events-resource.js [hunk 1]
```diff
@@ -1,91 +1,97 @@
...
+    createDraftRuleset({eventType, data}) {
+      return apiHandler.post(`events/${eventType}/rules/drafts`, data);
+    },
+    getAllDraftRuleset({eventType, id, fields = null, expand = null}) {
+      const params = {fields, expand};
+      return apiHandler.get(`events/${eventType}/rules/drafts/${id}`, params);
+    },
+    updateDraftRuleset({eventType, id, data}) {
+      return apiHandler.put(`events/${eventType}/rules/drafts/${id}`, data);
+    },
```
Comment (rensftw) [inline src/resources/events-resource.js:92]: The `data` parameter used to have a default value of `{}` before and it doesn't have it now - maybe this is contributing to the issue?

File: src/resources/events-resource.js [hunk 2]
```diff
@@ -1,91 +1,97 @@
...
+      sort = null,
+      fields = null,
+      expand = null,
+    }) {
+      const params = {limit, offset, filter, q, sort, fields, expand};
+      return apiHandler.getAll(`events/${eventType}/rules/drafts`, params);
+    },
+    createDraftRuleset({eventType, data}) {
+      return apiHandler.post(`events/${eventType}/rules/drafts`, data);
+    },
```
Comment (rensftw) [inline src/resources/events-resource.js:85]: `data` used to have a default value of `{}` - I haven't detected an issue with that yet, but idk if that's ok?

File: src/resources/events-resource.js [hunk 3]
```diff
@@ -1,91 +1,97 @@
...
+    getRulesVersionDetail({eventType, version, fields = null, expand = null}) {
+      const params = {fields, expand};
+      return apiHandler.get(
+        `events/${eventType}/rules/versions/${version}`,
+        params
+      );
+    },
+    getAllDraftRulesets({
+      eventType,
```
Comment (rensftw) [inline src/resources/events-resource.js:71]: `eventType` used to have a default value `null` - is this ok?

Comment (agualis) [reply src/resources/events-resource.js:71]: This is ok cause eventType is required.

File: src/resources/events-resource.js (hunk 2 above)
Comment (agualis) [reply src/resources/events-resource.js:85]: `data` should be required when you are creating. I'm going to investigate the error further.

**Comment by agualis** [comment]
@rensftw problem solved: we were missing `getDraftRuleset` function in customizations. Thanks for your help!

File: src/resources/events-resource.js [hunk 4]
```diff
@@ -1,92 +1,97 @@
...
+      return apiHandler.post(`events/${eventType}/rules/drafts`, data);
+    },
+    getDraftRuleset({eventType, id, fields = null, expand = null}) {
+      const params = {fields, expand};
+      return apiHandler.get(`events/${eventType}/rules/drafts/${id}`, params);
+    },
+    updateDraftRuleset({eventType, id, data}) {
+      return apiHandler.put(`events/${eventType}/rules/drafts/${id}`, data);
+    },
+    deleteDraftRuleset({eventType, id}) {
```
Comment (rensftw) [inline src/resources/events-resource.js:93]: We have changed the named export from `messageId` to `id` - wouldn't this break stuff?

File: src/resources/events-resource.js [hunk 5]
```diff
@@ -1,92 +1,97 @@
...
+      filter = null,
+      q = null,
+      sort = null,
+      fields = null,
+      expand = null,
+    }) {
+      const params = {limit, offset, filter, q, sort, fields, expand};
+      return apiHandler.getAll(`events/${eventType}/rules/drafts`, params);
+    },
+    createDraftRuleset({eventType, data}) {
```
Comment (rensftw) [inline src/resources/events-resource.js:83]: We have removed the default values - is this ok?
Used to be `createDraftRuleset({eventType = null, data = {}} = {})`

File: src/resources/events-resource.js [hunk 6]
```diff
@@ -1,92 +1,97 @@
...
+      return apiHandler.post(`events/${eventType}/rules/drafts`, data);
+    },
+    getDraftRuleset({eventType, id, fields = null, expand = null}) {
+      const params = {fields, expand};
+      return apiHandler.get(`events/${eventType}/rules/drafts/${id}`, params);
+    },
+    updateDraftRuleset({eventType, id, data}) {
```
Comment (rensftw) [inline src/resources/events-resource.js:90]: We have removed the default values - is this ok?

**Review by rensftw — APPROVED** [review]

**Pull request merged**